BetterPasswordPlugin.inc.php

```diff
@@ -158,26 +158,17 @@ function checkPassword($hookName, $args) {
 			return false;
 		}
 		if ($this->getSetting(CONTEXT_SITE, 'betterPasswordCheckBlacklist')) {
-			$badPassword = false;
 			$lowerPassword = strtolower($password);
-			foreach ($this->getBlacklists() as $filename) {
-				/* In case of out-of-memory error, break glass
-				$passwordfile = fopen($filename, "r");
-				while (!feof($passwordfile)) {
-					$disallowed = fgets($passwordfile);
-					if ($lowerPassword === $disallowed) {
-						$badPassword = true;
-						break;
-					}
-				}
-				fclose($passwordfile);
-				 */
-				$passwordfile = file_get_contents($filename);
-				if (strpos($passwordfile, $lowerPassword) !== false) {
-					$badPassword = true;
-					break;
-				}
-			}
+                        $shaPass = sha1($lowerPassword);
+                        $passwordHash = substr($shaPass,0,2);
+                        $blacklist = $this->blacklistSetting();
+                        if ($blacklist) {
+                            $cache = CacheManager::getManager()->getCache('badPasswords', $passwordHash, array($this, '_PasswordCacheMiss'));
+                            $badPassword = $cache->get($shaPass);
+                        }
+                        else {
+                            $badPassword = true;
```

Comment:
The new flow of is very clear. Nice!  This error condition should generate a message to the user distinct from the message "plugins.generic.betterPassword.validation.betterPasswordCheckBlacklist", however.  There isn't a good existing error message in the core product which we can reuse, unfortunately.  Please [create a new message](https://github.com/ulsdevteam/pkp-betterPassword/blob/0d50a147861d4982cc412635876f5293d3540e88/locale/en_US/locale.xml#L23) like:
```
	<message key="plugins.generic.betterPassword.validation.betterPasswordUnexpectedError">Your password could not be changed at this time.  Please try again later.</message>
```